docs/Learning-Environment-Design-Agents.md

```diff
@@ -390,20 +390,20 @@ impossible for the next decision. When the Agent is controlled by a
 neural network, the Agent will be unable to perform the specified action. Note
 that when the Agent is controlled by its Heuristic, the Agent will
 still be able to decide to perform the masked action. In order to mask an
-action, call the method `SetActionMask` on the optional `ActionMasker` argument of the `CollectObservation` method :
+action, call the method `SetMask` on the `DiscreteActionMasker` argument of the `CollectDiscreteActionMasks` method :
```

Comment:
```suggestion
action,  override the `Agent.CollectDiscreteActionMasks()` virtual method, and call `DiscreteActionMasker.SetMask()` in it:
```
I think the old wording wasn't obvious that the user needed to override the method.